### ISSUE TITLE
Broken instance after Upgrade 4.7 to 4.8 

### ISSUE THREAD

**Issue by uhlhosting** [wazuh/wazuh#24170]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0 | Wazuh indexer | Manager | Packages | Ubuntu 22.04 |

```
wazuh-dashboard.service - wazuh-dashboard
     Loaded: loaded (/etc/systemd/system/wazuh-dashboard.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2024-06-19 14:23:58 CEST; 43s ago
   Main PID: 3167719 (node)
      Tasks: 11 (limit: 933)
     Memory: 332.8M
        CPU: 11.542s
     CGroup: /system.slice/wazuh-dashboard.service
             └─3167719 /usr/share/wazuh-dashboard/node/bin/node --no-warnings --max-http-header-size=65536 --unhandled-rejections=warn /usr/share/wazuh-dashboard/src/cli/dist

Jun 19 14:24:20 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:20Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
Jun 19 14:24:23 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:23Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ResponseError]: Response Error"}
Jun 19 14:24:25 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:25Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ResponseError]: Response Error"}
Jun 19 14:24:28 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:28Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ResponseError]: Response Error"}
Jun 19 14:24:30 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:30Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ResponseError]: Response Error"}
Jun 19 14:24:33 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:33Z","tags":["error","opensearch","data"],"pid":3167719,"message":"[ResponseError]: Response Error"}
Jun 19 14:24:36 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:36Z","tags":["info","savedobjects-service"],"pid":3167719,"message":"Starting saved objects migrations"}
Jun 19 14:24:36 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:36Z","tags":["info","plugins-system"],"pid":3167719,"message":"Starting [52] plugins: [usageCollection,opensearchDashboardsUsageCollection,o>
Jun 19 14:24:37 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:37Z","tags":["listening","info"],"pid":3167719,"message":"Server running at [https://0.0.0.0:443](https://0.0.0.0/)"}
Jun 19 14:24:37 wazu opensearch-dashboards[3167719]: {"type":"log","@timestamp":"2024-06-19T12:24:37Z","tags":["info","http","server","OpenSearchDashboards"],"pid":3167719,"message":"http server running at [https://0.0.0.0:443](https://0.0.0.0/)"}
```

```
wazuh-indexer.service - Wazuh-indexer
     Loaded: loaded (/lib/systemd/system/wazuh-indexer.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2024-06-19 14:24:21 CEST; 20s ago
       Docs: https://documentation.wazuh.com/
   Main PID: 3167788 (java)
      Tasks: 133 (limit: 933)
     Memory: 5.0G
        CPU: 1min 54.441s
     CGroup: /system.slice/wazuh-indexer.service
             └─3167788 /usr/share/wazuh-indexer/jdk/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.negative.ttl=10 -XX:+AlwaysPreTouch -Xss1m -Djava.awt.headless=true -Dfile.encoding=UTF-8>

Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:206)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:204)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:242)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:849)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:282)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:245)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]:         at java.base/java.lang.Thread.run(Thread.java:833)
Jun 19 14:24:22 wazu systemd-entrypoint[3167788]: For complete error details, refer to the log at /var/log/wazuh-indexer/wazuh-cluster.log
```

```
INFO: No current API selected
INFO: Getting API hosts...
INFO: API hosts found: 1
INFO: Checking API host id [default]...
INFO: Could not connect to API id [default]: 3099 - ERROR3099 - Wazuh not ready yet
INFO: Removed [navigate] cookie
ERROR: No API available to connect
```

```
INFO: Index pattern id in cookie: yes [wazuh-alerts-*]
INFO: Getting list of valid index patterns...
INFO: Valid index patterns found: 1
INFO: Found default index pattern with title [wazuh-alerts-*]: yes
INFO: Checking the app default pattern exists: id [wazuh-alerts-*]...
INFO: Default pattern with id [wazuh-alerts-*] exists: yes
ACTION: Default pattern id [wazuh-alerts-*] set as default index pattern
INFO: Checking the index pattern id [wazuh-alerts-*] exists...
INFO: Index pattern id exists [wazuh-alerts-*]: yes
INFO: Index pattern id in cookie: yes [wazuh-alerts-*]
INFO: Checking if the index pattern id [wazuh-alerts-*] exists...
INFO: Index pattern id [wazuh-alerts-*] found: yes title [wazuh-alerts-*]
INFO: Checking if exists a template compatible with the index pattern title [wazuh-alerts-*]
INFO: Template found for the selected index-pattern title [wazuh-alerts-*]: yes
INFO: Index pattern id in cookie: [wazuh-alerts-*]
INFO: Getting index pattern data [wazuh-alerts-*]...
INFO: Index pattern data found: [yes]
INFO: Refreshing index pattern fields: title [wazuh-alerts-*], id [wazuh-alerts-*]...
ACTION: Refreshed index pattern fields: title [wazuh-alerts-*], id [wazuh-alerts-*]
INFO: Getting settings...
INFO: Check Wazuh dashboard setting [timeline:max_buckets]: 200000
INFO: App setting [timeline:max_buckets]: 200000
INFO: Settings mismatch [timeline:max_buckets]: no
INFO: Getting settings...
INFO: Check Wazuh dashboard setting [metaFields]: ["_source","_index"]
INFO: App setting [metaFields]: ["_source","_index"]
INFO: Settings mismatch [metaFields]: no
INFO: Getting settings...
INFO: Check Wazuh dashboard setting [timepicker:timeDefaults]: {"from":"now-24h","to":"now"}
INFO: App setting [timepicker:timeDefaults]: "{\"from\":\"now-24h\",\"to\":\"now\"}"
INFO: Settings mismatch [timepicker:timeDefaults]: no
```

**Comment by sjansen1** [comment]
Exactly the same issue after updating wazuh from 4.7.5-1 to 4.8.0-1. Exactly the same errors in logs.

**Comment by genseirin** [comment]
Seems to be the same here. "ERROR: No API available to connect" with 4.8.0. Haven't changed anything in the configuration.

**Comment by sjansen1** [comment]
I was unable to solve this by myself. I just wiped the system and reinstalled it with 4.8.0, this time i used to opertunity to use Ubuntu 24.04.. That was successful. Okay for homelab, not great if this would be a production workload.

**Comment by genseirin** [comment]
I'd be happy to find another solution than wiping the entire system. If a fresh install solves the issue, the old configuration could be incompatible with the new version.

**Comment by lachlan2k** [comment]
Fixed for me by replacing `localhost` with `127.0.0.1` in `/etc/wazuh-dashboard/opensearch_dashboards.yml`

**Comment by sjansen1** [comment]
For me it just worked on time after wipe and reinstall, on the next restart its broken again.
